src/core/overflowsafe_type.hpp

```diff
@@ -104,25 +115,30 @@ class OverflowSafeInt
 
 	/* Operators for addition and subtraction */
 	inline OverflowSafeInt  operator +  (const OverflowSafeInt& other) const { OverflowSafeInt result = *this; result += other; return result; }
-	inline OverflowSafeInt  operator +  (const int              other) const { OverflowSafeInt result = *this; result += (int64)other; return result; }
-	inline OverflowSafeInt  operator +  (const uint             other) const { OverflowSafeInt result = *this; result += (int64)other; return result; }
+	template <class I, class = typename std::enable_if<std::is_integral<I>::value>::type>
+	inline OverflowSafeInt& operator += (const I                other)       { return *this += OverflowSafeInt(other); }
+	template <class I, class = typename std::enable_if<std::is_integral<I>::value>::type>
+	inline OverflowSafeInt  operator +  (const I                other) const { return *this +  OverflowSafeInt(other); }
```

Comment:
(I think) we need to typecast, in order to be sure `other` is of a compatible type to `this->m_value`.